src/Fluid/Flu_Prepare.cpp

```diff
@@ -159,9 +228,16 @@ void Flu_Prepare( const int lv, const double PrepTime,
 
             for (int v=0; v<FLU_NIN; v++)
             {
-               if (  !Aux_IsFinite( fluid[v] )  )
+               bool isDataInvalid = !Aux_IsFinite( fluid[v] );
+
+//             check for negative densities
+#              if ( MODEL == ELBDM )
+               isDataInvalid |= (v == DENS) && fluid[DENS] < 0;
```

Comment:
[Bug] Should check density only on the fluid level but not the wave level.

Reply:
@hyschive 
Fixed.